vulnerabilities/views.py

```diff
@@ -116,6 +116,8 @@ def get_queryset(self):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
+        weaknesses = self.object.weaknesses.all()
+        weaknesses_present_in_db = [weakness for weakness in weaknesses if weakness.weakness]
```

Comment:
`[weakness for weakness in weaknesses if weakness.weakness]`
This naming is not great.

Reply:
done

Reply:
I was trying to point out a design issue. We can change variable names, but we're simply hiding the underlying issues.

Reply:
@tdruez In that case, we first need a migration, that will remove the CWEs that are stored in advisories and CWE table. Then overwrite save, to verify if a CWE exists on CWE2 side before storing it into DB again.

Reply:
@ziadhany FYI ^^

Reply:
@TG1999 
According to cwe.mitre.org:  [CWE-189](https://cwe.mitre.org/data/definitions/189.html)  is CWE CATEGORY  .
`Vulnerability Mapping Notes : (this CWE ID must not be used to map to real-world vulnerabilities)`
so I think we should remove all CWE that have this Note .